### PR TITLE
Register DEFAULT group on agent start

### PR DESCRIPTION
**Pull request by sven-rosenzweig**
From upstream commit 4d3a2747 fixing bug (#1968606), registration of the group DEFAULT does not happen automatically anymore.

If not loaded, agent fails with 'oslo_config.cfg.NoSuchOptError: no such option api_extensions_path in group [DEFAULT]'

### CODE REVIEW

**Review by sebageek — APPROVED** [review]

File: asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py [hunk 1]
```diff
@@ -88,6 +88,7 @@
 def main(manager='asr1k_neutron_l3.plugins.l3.agents.asr1k_l3_agent.L3ASRAgentWithStateReport'):
     asr1k_config.register_common_opts()
     asr1k_config.register_l3_opts()
+    common_config.register_core_common_config_opts()
```
Comment (sebageek) [inline asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py]: Are you sure `common_config` is the right module? We import it as `from neutron.common import config as common_config` but the method is defined in `neutron.conf.common` (for stable/yoga).

**Review by sebageek — CHANGES_REQUESTED** [review]
Is the import correct?

File: asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py (hunk 1 above)
Comment (sven-rosenzweig) [reply asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py]: Yes, import is right (for stable/2024.1). (https://github.com/sapcc/neutron/blob/0aa154b5ce9dc8da73309fb212843a2b69b68696/neutron/common/config.py#L52). Fixing the [bug](https://bugs.launchpad.net/neutron/+bug/1968606) makes this necessary.

Comment (sebageek) [reply asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py]: The commit you're referencing is not part of `stable/yoga-m3`. I can't find the method in the referenced file in `stable/yoga-m3`, meaning if we're going to merge it we're going to break the asr driver on yoga release - oooor I looked wrong and it's imported from somewhere? But if I'm right we need to either fix the import for yoga or we only fix this in Caracal, meaning you'd need to open this PR against `stable/2024.1-m3`.

Comment (sven-rosenzweig) [reply asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py]: Thx for pointing me to this. I wanted to open the PR against `stable/2024.1-m3`.

Comment (sebageek) [reply asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py]: As per our call let's use `register_common_config_options()` from `neutron.common.config`.

**Review by sebageek — APPROVED** [review]

**Review by swagner-de — APPROVED** [review]

**Pull request merged**